clients/drcachesim/tracer/output.cpp

```diff
@@ -445,15 +445,21 @@ open_new_thread_file(void *drcontext, ptr_int_t window_num)
     return opened_new_file;
 }
 
+/* Returns the size of just the thread header, but appends both the thread header
+ * and the buffer header.
+ */
 static size_t
-prepend_offline_thread_header(void *drcontext)
+prepend_offline_thread_header(void *drcontext, ptr_int_t window = -1)
 {
     DR_ASSERT(op_offline.get_value());
     /* Write initial headers at the top of the first buffer. */
     per_thread_t *data = (per_thread_t *)drmgr_get_tls_field(drcontext, tls_idx);
     size_t size = reinterpret_cast<offline_instru_t *>(instru)->append_thread_header(
         data->buf_base, dr_get_thread_id(drcontext), get_file_type());
-    BUF_PTR(data->seg_base) = data->buf_base + size + buf_hdr_slots_size;
+    BUF_PTR(data->seg_base) = data->buf_base + size;
+    BUF_PTR(data->seg_base) += append_unit_header(
```

Comment:
Why do we need to add this here? If we don't add it here, then we won't need the if(!prepended) condition checks below.

Reply:
This append needs to be done at some point for every call to this function prepend_offline_thread_header() which is why it is logically bundled here.  If it is moved from here it needs to be added to every call site.  Maybe that is simpler than the conditional: there are only two call sites.  The risk is a future new call that forgets to also add the unit header but maybe that is still worthwhile.  OK, will do.

Reply:
Actually I end up still needing the prepended return value and check as I don't know whether to add the header to the current pointer or the base here.  Here we could set the pointer to the base before calling set_local_window; up above on an empty buffer though we still need the return value as we again don't know where to write the unit header.  It's less clear whether this is better than before but it's already in there so I guess we'll go with it, after some testing.